### PR TITLE
S06 Sidebar

### PR DESCRIPTION
**Pull request by No1Nero**

### CODE REVIEW

File: src/components/MobileModalLayout/MobileModalLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    return (
+        <>
+        <div className='mobile_modal_layout'></div>
+        <div className='mobile_modal_layout_content'>
+            <img onClick={onHandleFiltersShown} className='mobile_modal_layout_cross_icon' alt='Cross Icon' src={cross_icon} />
+            {children}
+        </div>
+        </>
+    );
+};
```
Comment (Yuliia143) [inline src/components/MobileModalLayout/MobileModalLayout.tsx:19]: Need empty line

File: src/components/Searchbar/Searchbar.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ export function Searchbar() {
                 <input onFocus={() => toggleProductsDropdown(true)} onChange={handleChangeSearchBar} value={filterSearchBar} type='text' className='search_bar_input' placeholder='Search Products...' />
                 <button onClick={clearSearchbar} className='search_bar_button'><img alt='Cross button' src={cross_icon} /></button>
                 <button className='search_bar_button'><img alt='Search button' src={search_icon} /></button>
-                {pathname !== routerConstants.PRODUCTS && isFocusOnInput && filterSearchBar.length >= MIN_SEARCHBAR_LENGTH_FOR_SEARCH &&
+                {pathname !== routerConstants.PRODUCTS && isFocusOnInput && filterSearchBar.trim().length >= MIN_SEARCHBAR_LENGTH_FOR_SEARCH &&
```
Comment (Yuliia143) [inline src/components/Searchbar/Searchbar.tsx:53]: Better to have some variable for this condition for better understanding what it means

File: src/components/Sidebar/SidebarBrandFilter/SidebarBrandFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      <ul className="sidebar_brand_filter_list">
+          <li className="sidebar_brand_filter_list_item">
+              <CustomCheckbox checked={!filterBrand.length} onChange={() => setFilterBrand([])} />
+              <p onClick={() => setFilterBrand([])} className="sidebar_brand_filter_brand_title">All brands</p>
+          </li>
+          {brandsForCategory?.map(({ brands }) =>
+          brands.map((brand) => (
+              <li className="sidebar_brand_filter_list_item" key={brand}>
+                  <CustomCheckbox checked={filterBrand.includes(brand)} onChange={() => setFilterBrand(brand)} />
+                  <p onClick={() => setFilterBrand(brand)} className="sidebar_brand_filter_brand_title">{brand}</p>
```
Comment (Yuliia143) [inline src/components/Sidebar/SidebarBrandFilter/SidebarBrandFilter.tsx:34]: Isn't it better to have onClick for li ?

File: src/components/Sidebar/SidebarCategoryFilter/SidebarCategoryFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  return (
+    <ul className="sidebar_category_filter_list">
+      <li onClick={setAllCategories} className="sidebar_category_filter_list_item">
+        <div className="sidebar_category_filter_radio_item">
+          <input
+            className="sidebar_category_filter_radio_button"
+            onChange={setAllCategories}
+            checked={filterCategory === ""}
```
Comment (Yuliia143) [inline src/components/Sidebar/SidebarCategoryFilter/SidebarCategoryFilter.tsx:32]: ?

File: src/components/Sidebar/SidebarCategoryFilter/SidebarCategoryFilter.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            type="radio"
+            name="Category"
+            value={""}
+          />
+          <p className={filterCategory === "" ? "sidebar_category_filter_checked_category_title": "sidebar_category_filter_category_title"}>All categories</p>
```
Comment (Yuliia143) [inline src/components/Sidebar/SidebarCategoryFilter/SidebarCategoryFilter.tsx:37]: filterCategory === "" ?

File: src/components/Sidebar/SidebarBrandFilter/SidebarBrandFilter.tsx (hunk 3 above)
Comment (No1Nero) [reply src/components/Sidebar/SidebarBrandFilter/SidebarBrandFilter.tsx:34]: If I set onClick to li, the custom checkbox doesn't respond to clicks

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**